cwms-data-api/src/test/java/cwms/cda/data/dao/watersupply/WaterContractDaoTestIT.java

```diff
@@ -0,0 +1,386 @@
+        });
+        cleanupUserRoutine(newUser);
+    }
+
+    @Test
+    void testStoreAndRetrieveWaterContractList() throws Exception {
+        WaterUserContract contract = buildTestWaterContract("Test retrieve", false);
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            WaterContractDao dao = new WaterContractDao(ctx);
+            dao.storeWaterUser(contract.getWaterUser(), true);
+            dao.storeWaterContract(contract, true, true);
+            List<WaterUserContract> retrievedContract = dao.getAllWaterContracts(contract.getWaterUser().getProjectId(), contract.getWaterUser().getEntityName());
+            WaterUserContract result = retrievedContract.get(0);
+            DTOMatch.assertMatch(contract, result);
+        });
+        cleanupContractRoutine(contract);
+        cleanupUserRoutine(contract.getWaterUser());
+    }
+
+    @Test
+    void testStoreAndRetrieveWaterContractTypeList() throws Exception {
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            WaterContractDao dao = new WaterContractDao(ctx);
+            List<LookupType> contractType = new ArrayList<>();
+            contractType.add(new LookupType.Builder()
+                    .withTooltip("Test Tooltip")
+                    .withActive(true)
+                    .withDisplayValue("Test Display Value")
+                    .withOfficeId(OFFICE_ID).build());
+            contractType.add(new LookupType.Builder()
+                    .withTooltip("Test Tooltip 2")
+                    .withActive(true)
+                    .withDisplayValue("Test Display Value 2")
+                    .withOfficeId(OFFICE_ID).build());
+            dao.storeWaterContractTypes(contractType, false);
+            List<LookupType> results = dao.getAllWaterContractTypes(OFFICE_ID);
+            DTOMatch.assertMatch(contractType.get(0), results.get(0));
+            DTOMatch.assertMatch(contractType.get(1), results.get(1));
+        });
+    }
+
+    @Test
+    void testRenameWaterUser() throws Exception {
+        WaterUser newUser = buildTestWaterUser("TEST RENAME USER");
+        WaterUser updatedUser = buildTestWaterUser("TEST RENAME USER 2");
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            WaterContractDao dao = new WaterContractDao(ctx);
+            dao.storeWaterUser(newUser, true);
+            dao.renameWaterUser(newUser.getEntityName(), updatedUser.getEntityName(), newUser.getProjectId());
+            WaterUser retrievedUser = dao.getWaterUser(newUser.getProjectId(), updatedUser.getEntityName());
+            DTOMatch.assertMatch(updatedUser, retrievedUser);
```

Comment:
check to make sure the old water user is gone

Reply:
Added check for water user no longer in DB